rest/blip_client_test.go

```diff
@@ -65,19 +70,153 @@ type BlipTesterClient struct {
 	nonCollectionAwareClient *BlipTesterCollectionClient
 }
 
+func (c *BlipTesterCollectionClient) getClientDocForSeq(seq clientSeq) (*clientDoc, bool) {
+	c.seqLock.RLock()
+	defer c.seqLock.RUnlock()
+	doc, ok := c._seqStore[seq]
+	return doc, ok
+}
+
+// OneShotDocsSince is an iterator that yields client sequence and document pairs that are newer than the given since value.
+func (c *BlipTesterCollectionClient) OneShotDocsSince(since clientSeq) iter.Seq2[clientSeq, *clientDoc] {
+	return func(yield func(clientSeq, *clientDoc) bool) {
+		c.seqLock.Lock()
+		seqLast := c._seqLast
+		for c._seqLast <= since {
+			// block until new seq
+			c.TB().Logf("OneShotDocsSince: since=%d, _seqLast=%d - waiting for new sequence", since, c._seqLast)
+			c._seqCond.Wait()
```

Comment:
I'm worried about the case where a test assertion fails in the middle of the iteration, or a test times out? A sync.Cond will block forever, and we can't cancel with a context. For the changes feed we use the idiom of Broadcast() and then checking under wait to see if the context was cancelled.

I'm not sure this is a problem, but I am worried about a failing test failing to take down its rest tester, either as a data leak or failing to exit appropriately.

Reply:
I think maybe calling Broadcast() on close and doing a ctx check to return would resolve this?  I'm not overly familiar with sync.Cond idioms

Reply:
FYI, I still need to do this and the other context/close investigation we talked about.